controllers/k8sgpt_controller.go

```diff
@@ -179,18 +179,21 @@ func (r *K8sGPTReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 
 	if deployment.Status.ReadyReplicas > 0 {
 
-		// Check the version of the deployment image matches the version set in the K8sGPT CR
+		// Check the repo and version of the deployment image matches the repo and version set in the K8sGPT CR
 		imageURI := deployment.Spec.Template.Spec.Containers[0].Image
 
 		image := strings.Split(imageURI, ":")
-		imageRepository := image[0]
-		imageVersion := image[1]
+		imageRepository := strings.Join(image[0:len(image)-1], ":")
```

Comment:
if you don't specify a tag in your custom registry e.g `"host:8080/path/image"` the split will be wrong, right ?

Reply:
If I don't specify a tag code will `panic: runtime error: index out of range [1] with length 1` here I believe:
https://github.com/k8sgpt-ai/k8sgpt-operator/blob/2eae459132cec106a7b7933e419f91269e9fd8eb/controllers/k8sgpt_controller.go#L187

I will fix that somehow.

Reply:
It's about how it works now. What tag should we use it is not specified? `:latest`?

Reply:
Yes, latest makes sense